### PR TITLE
Wire up `oneversion` in Java toolchain configurations

### PR DESCRIPTION
**Pull request by fmeum**

### CODE REVIEW

**Comment by fmeum** [comment]
@cushon Since `--experimental_one_version_enforcement` is `OFF` by default, I think that it won't hurt to register it in the default toolchain configuration, even with no way to allowlist existing targets yet.

**Review by cushon — APPROVED** [review]

**Comment by fmeum** [comment]
@keertk After this has been merged, could we get a new java_tools/rules_java release to pick this up together with the turbine update and https://github.com/bazelbuild/bazel/commit/407ed1dd17a0aa442816ac41efb6d81d8cf14ae9?

**Comment by keertk** [comment]
@fmeum sure, tracked here: https://github.com/bazelbuild/java_tools/issues/91

File: MODULE.bazel [hunk 1]
```diff
@@ -14,6 +14,7 @@ bazel_dep(name = "bazel_skylib", version = "1.6.1")
 # Required by @remote_java_tools, which is loaded via module extension.
 bazel_dep(name = "rules_proto", version = "4.0.0")
 bazel_dep(name = "rules_license", version = "0.0.3")
+bazel_dep(name = "abseil-cpp", version = "20240116.2", repo_name = "com_google_absl")
```
Comment (hvadehra) [inline MODULE.bazel:17]: @fmeum Why was this new dep required in this change?

Comment (fmeum) [reply MODULE.bazel:17]: This is a dep of oneversion when built from source.

Comment (hvadehra) [reply MODULE.bazel:17]: I see, thanks.

**Pull request closed without merging**